web/static/js/app.js

```diff
@@ -17,22 +17,57 @@ class App {
     socket.onError( ev => console.log("ERROR", ev) )
     socket.onClose( e => console.log("CLOSE", e))
 
-    var chan = socket.channel("rooms:lobby", {})
+    var chan = socket.channel("__absinthe__:control", {})
+
+    let subscription = {
+      query: `
+      subscription Messages {
+        message(room: "lobby") {
+          body
+          author { name }
+        }
+      }
+      `,
+      variables: {}
+    }
+
     chan.join().receive("ignore", () => console.log("auth error"))
-               .receive("ok", () => console.log("join ok"))
-               .after(10000, () => console.log("Connection interruption"))
-    chan.onError(e => console.log("something went wrong", e))
-    chan.onClose(e => console.log("channel closed", e))
+      .receive("ok", () => {
+        console.log("join ok")
+
+        chan.push("doc", subscription)
+          .receive("ok", (msg) => console.log("subscription created", msg) )
+          .receive("error", (reasons) => console.log("subscription failed", reasons) )
+          .receive("timeout", () => console.log("Networking issue...") )
+      })
+      .after(10000, () => console.log("Connection interruption"))
+
 
     $input.off("keypress").on("keypress", e => {
       if (e.keyCode == 13) {
-        chan.push("new:msg", {user: $username.val(), body: $input.val()})
+        chan.push("doc", {
+          query:
+          `
+          mutation SendMessage($body: String!, $user: String!) {
+            sendMessage(room: "lobby", body: $body, user: $user) {
+              __typename
+            }
+          }
+          `,
+          variables: {
+            body: $input.val(),
+            user: $username.val()
+          }
+        })
+        .receive("ok", (msg) => console.log("mutation succeeded", msg) )
+        .receive("error", (reasons) => console.log("mutation failed", reasons) )
+        .receive("timeout", () => console.log("Networking issue...") )
         $input.val("")
       }
     })
 
-    chan.on("new:msg", msg => {
-      $messages.append(this.messageTemplate(msg))
+    chan.on("subscription:data", msg => {
```

Comment:
@benwilson512 

Today I got [the example chat application working with apollo subscriptions](https://github.com/vic/apollo-phoenix-websocket/pull/9#issuecomment-296416979), the only change worth of mentioning is [this one](https://github.com/vic/phoenix_chat_example/commit/addafdb6d9a7d47e2c4c9a361aaef8e9aa7b01c3#diff-27bd4685ee7a71d148351102b0d36a79R23) so that updates get sent (and most importantly,  the client subscribes to) different events per gql subscription, instead of always getting them on a single `subscription:data` event.